Previously, gene expression features in ADNI database was used as it is. After coming across the paper "https://www.nature.com/articles/s41598-020-60595-1" it felt necessary to perform following pre-processing before using the data for our use. 
1. Only keep the high quality gene expression data for subjects with RIN (RNA integrity number) value is >6.9 
2. To reduce the background noise of ANDI, exclude the probes with an intensity value ≤ the median of all gene expression values in 100 or more samples.
3. If there were multiple probes annotated in one gene, then the median value of those was selected.

In [12]:
#let's load the already prepared gene expression data
import pandas as pd
import numpy as np
import os

data_prep_path = '/Users/subashkhanal/Desktop/MLforAlzheimers/data/ADNI/genetics/data_prep'
stats_path = '/Users/subashkhanal/Desktop/MLforAlzheimers/data/ADNI/genetics/stats'

In [13]:
df = pd.read_csv(os.path.join(data_prep_path,'ADNI_Imaging_Gene_Expression_MMSE_geneLabeled_Combined.csv'),low_memory=False)

In [14]:
df = df.T
df

0     1        2         3     4         5         6      \
PTID        COLPROT   RIN  VISCODE  EXAMDATE   AGE  PTGENDER  PTEDUCAT   
Gene            NaN   999      NaN       NaN   NaN       NaN       NaN   
116_S_1249   ADNIGO   7.7      m48   2/23/11  70.8    Female        15   
037_S_4410    ADNI2   7.6       bl    1/4/12  69.1    Female        14   
006_S_4153    ADNI2   7.2       bl    9/9/11  79.3      Male        20   
...             ...   ...      ...       ...   ...       ...       ...   
009_S_2381   ADNIGO   6.6       bl   5/31/11  69.1      Male        16   
053_S_4557    ADNI2   6.8       bl   3/15/12  83.3    Female        12   
073_S_4300    ADNI2   6.2       bl  11/16/11  80.6    Female        16   
041_S_4014    ADNI2   5.8       bl   5/31/11  80.7      Male        16   
007_S_0101    ADNI2   6.7      m72   2/16/12  73.6      Male        18   

                      7         8      9      ...            49408  \
PTID               PTETHCAT  PTRACCAT  APOE4  ...  AFFX-r2-TagH_at   
Gene                    NaN       NaN    NaN  ...              NaN   
116_S_1249  Not Hisp/Latino     White      1  ...            2.355   
037_S_4410  Not Hisp/Latino     White      0  ...              2.1   
006_S_4153  Not Hisp/Latino     White      1  ...            2.165   
...                     ...       ...    ...  ...              ...   
009_S_2381  Not Hisp/Latino     White      2  ...            2.383   
053_S_4557  Not Hisp/Latino     White      0  ...            2.046   
073_S_4300  Not Hisp/Latino     White      0  ...            2.129   
041_S_4014  Not Hisp/Latino     White      0  ...            2.331   
007_S_0101  Not Hisp/Latino     White      2  ...            2.145   

                         49409               49410               49411  \
PTID        AFFX-r2-TagIN-3_at  AFFX-r2-TagIN-5_at  AFFX-r2-TagIN-M_at   
Gene                       NaN                 NaN                 NaN   
116_S_1249               2.624                2.01               2.906   
037_S_4410                2.82               1.726               2.465   
006_S_4153               2.455                1.84               2.681   
...                        ...                 ...                 ...   
009_S_2381               2.937               1.964               2.662   
053_S_4557               2.585               1.997               2.732   
073_S_4300               2.464               1.924               2.613   
041_S_4014               2.695               2.108               2.661   
007_S_0101               2.757               1.905               2.619   

                        49412              49413              49414  \
PTID        AFFX-r2-TagJ-3_at  AFFX-r2-TagJ-5_at  AFFX-r2-TagO-3_at   
Gene                      NaN                NaN                NaN   
116_S_1249              2.463               2.05               2.06   
037_S_4410               2.26              1.933              1.717   
006_S_4153              2.251              1.985               1.77   
...                       ...                ...                ...   
009_S_2381              2.478              2.017              1.886   
053_S_4557              2.181              1.885              1.742   
073_S_4300              2.269              1.807              1.781   
041_S_4014              2.383              2.188               1.74   
007_S_0101              2.324              1.962              1.906   

                        49415              49416              49417  
PTID        AFFX-r2-TagO-5_at  AFFX-r2-TagQ-3_at  AFFX-r2-TagQ-5_at  
Gene                      NaN                NaN                NaN  
116_S_1249              1.858              2.028              2.162  
037_S_4410              2.208              2.058              1.882  
006_S_4153              2.184              2.007              2.134  
...                       ...                ...                ...  
009_S_2381              2.371              1.997    

In [15]:
df.columns = df.iloc[0]
df = df.drop(df.index[[0]])
df

PTID       COLPROT  RIN VISCODE  EXAMDATE   AGE PTGENDER PTEDUCAT  \
Gene           NaN  999     NaN       NaN   NaN      NaN      NaN   
116_S_1249  ADNIGO  7.7     m48   2/23/11  70.8   Female       15   
037_S_4410   ADNI2  7.6      bl    1/4/12  69.1   Female       14   
006_S_4153   ADNI2  7.2      bl    9/9/11  79.3     Male       20   
116_S_1232  ADNIGO  6.8     m48   2/16/11  72.1   Female       14   
...            ...  ...     ...       ...   ...      ...      ...   
009_S_2381  ADNIGO  6.6      bl   5/31/11  69.1     Male       16   
053_S_4557   ADNI2  6.8      bl   3/15/12  83.3   Female       12   
073_S_4300   ADNI2  6.2      bl  11/16/11  80.6   Female       16   
041_S_4014   ADNI2  5.8      bl   5/31/11  80.7     Male       16   
007_S_0101   ADNI2  6.7     m72   2/16/12  73.6     Male       18   

PTID               PTETHCAT PTRACCAT APOE4  ... AFFX-r2-TagH_at  \
Gene                    NaN      NaN   NaN  ...             NaN   
116_S_1249  Not Hisp/Latino    White     1  ...           2.355   
037_S_4410  Not Hisp/Latino    White     0  ...             2.1   
006_S_4153  Not Hisp/Latino    White     1  ...           2.165   
116_S_1232  Not Hisp/Latino    White     0  ...           2.094   
...                     ...      ...   ...  ...             ...   
009_S_2381  Not Hisp/Latino    White     2  ...           2.383   
053_S_4557  Not Hisp/Latino    White     0  ...           2.046   
073_S_4300  Not Hisp/Latino    White     0  ...           2.129   
041_S_4014  Not Hisp/Latino    White     0  ...           2.331   
007_S_0101  Not Hisp/Latino    White     2  ...           2.145   

PTID       AFFX-r2-TagIN-3_at AFFX-r2-TagIN-5_at AFFX-r2-TagIN-M_at  \
Gene                      NaN                NaN                NaN   
116_S_1249              2.624               2.01              2.906   
037_S_4410               2.82              1.726              2.465   
006_S_4153              2.455               1.84              2.681   
116_S_1232              2.599              1.837              2.713   
...                       ...                ...                ...   
009_S_2381              2.937              1.964              2.662   
053_S_4557              2.585              1.997              2.732   
073_S_4300              2.464              1.924              2.613   
041_S_4014              2.695              2.108              2.661   
007_S_0101              2.757              1.905              2.619   

PTID       AFFX-r2-TagJ-3_at AFFX-r2-TagJ-5_at AFFX-r2-TagO-3_at  \
Gene                     NaN               NaN               NaN   
116_S_1249             2.463              2.05              2.06   
037_S_4410              2.26             1.933             1.717   
006_S_4153             2.251             1.985              1.77   
116_S_1232             2.158             1.916             1.878   
...                      ...               ...               ...   
009_S_2381             2.478             2.017             1.886   
053_S_4557             2.181             1.885             1.742   
073_S_4300             2.269             1.807             1.781   
041_S_4014             2.383             2.188              1.74   
007_S_0101             2.324             1.962             1.906   

PTID       AFFX-r2-TagO-5_at AFFX-r2-TagQ-3_at AFFX-r2-TagQ-5_at  
Gene                     NaN               NaN               NaN  
116_S_1249             1.858             2.028             2.162  
037_S_4410             2.208             2.058             1.882  
006_S_4153             2.184             2.007             2.134  
116_S_1232             2.163             2.185             2.099  
...                      ...               ...               ...  
009_S_2381             2.371             1.997             1.996  
053_S_4557             2.177             2.176             2.151  
073_S_4300              2.11             1.875             1.915  
041_S_4014             2.5

In [16]:
df.columns

Index(['COLPROT', 'RIN', 'VISCODE', 'EXAMDATE', 'AGE', 'PTGENDER', 'PTEDUCAT',
       'PTETHCAT', 'PTRACCAT', 'APOE4',
       ...
       'AFFX-r2-TagH_at', 'AFFX-r2-TagIN-3_at', 'AFFX-r2-TagIN-5_at',
       'AFFX-r2-TagIN-M_at', 'AFFX-r2-TagJ-3_at', 'AFFX-r2-TagJ-5_at',
       'AFFX-r2-TagO-3_at', 'AFFX-r2-TagO-5_at', 'AFFX-r2-TagQ-3_at',
       'AFFX-r2-TagQ-5_at'],
      dtype='object', name='PTID', length=49418)

In [23]:
df['RIN'] = df['RIN'].astype(float)

In [24]:
df = df[df['RIN']>6.9]

In [25]:
df

PTID       COLPROT    RIN VISCODE  EXAMDATE   AGE PTGENDER PTEDUCAT  \
Gene           NaN  999.0     NaN       NaN   NaN      NaN      NaN   
116_S_1249  ADNIGO    7.7     m48   2/23/11  70.8   Female       15   
037_S_4410   ADNI2    7.6      bl    1/4/12  69.1   Female       14   
006_S_4153   ADNI2    7.2      bl    9/9/11  79.3     Male       20   
099_S_4205   ADNI2    7.9      bl   11/4/11  81.4   Female       18   
...            ...    ...     ...       ...   ...      ...      ...   
052_S_0951  ADNIGO    7.1     m48  12/17/10  84.8     Male       18   
014_S_4401   ADNI2    7.0      bl  12/15/11  67.5   Female       20   
036_S_4562   ADNI2    7.1      bl   4/12/12  63.6     Male       18   
018_S_4696   ADNI2    8.2      bl    5/9/12    73   Female       16   
128_S_2130  ADNIGO    7.2      bl   1/19/11  76.4     Male       16   

PTID               PTETHCAT PTRACCAT APOE4  ... AFFX-r2-TagH_at  \
Gene                    NaN      NaN   NaN  ...             NaN   
116_S_1249  Not Hisp/Latino    White     1  ...           2.355   
037_S_4410  Not Hisp/Latino    White     0  ...             2.1   
006_S_4153  Not Hisp/Latino    White     1  ...           2.165   
099_S_4205  Not Hisp/Latino    White     0  ...           1.973   
...                     ...      ...   ...  ...             ...   
052_S_0951  Not Hisp/Latino    White     0  ...           2.204   
014_S_4401  Not Hisp/Latino    White     1  ...            2.21   
036_S_4562  Not Hisp/Latino    White     2  ...           2.422   
018_S_4696  Not Hisp/Latino    White     1  ...           2.157   
128_S_2130  Not Hisp/Latino    White     0  ...           2.283   

PTID       AFFX-r2-TagIN-3_at AFFX-r2-TagIN-5_at AFFX-r2-TagIN-M_at  \
Gene                      NaN                NaN                NaN   
116_S_1249              2.624               2.01              2.906   
037_S_4410               2.82              1.726              2.465   
006_S_4153              2.455               1.84              2.681   
099_S_4205              2.544              1.909              2.548   
...                       ...                ...                ...   
052_S_0951              2.778              1.884              2.733   
014_S_4401              2.291              2.079              2.873   
036_S_4562              2.944              1.694              2.527   
018_S_4696              2.622              1.775              2.823   
128_S_2130              3.092              1.816              2.742   

PTID       AFFX-r2-TagJ-3_at AFFX-r2-TagJ-5_at AFFX-r2-TagO-3_at  \
Gene                     NaN               NaN               NaN   
116_S_1249             2.463              2.05              2.06   
037_S_4410              2.26             1.933             1.717   
006_S_4153             2.251             1.985              1.77   
099_S_4205             2.266             2.077             1.838   
...                      ...               ...               ...   
052_S_0951             2.402             1.895             1.892   
014_S_4401             2.192             1.851             1.996   
036_S_4562             2.482               1.9             1.768   
018_S_4696             2.382             1.931             1.778   
128_S_2130             2.323             1.777             1.858   

PTID       AFFX-r2-TagO-5_at AFFX-r2-TagQ-3_at AFFX-r2-TagQ-5_at  
Gene                     NaN               NaN               NaN  
116_S_1249             1.858             2.028             2.162  
037_S_4410             2.208             2.058             1.882  
006_S_4153             2.184             2.007             2.134  
099_S_4205             2.085             1.941             1.883  
...                      ...               ...               ...  
052_S_0951              2.39             1.895             2.203  
014_S_4401             2.104              1.98              1.96  
036_S_4562             2.103             2.081             2.124  
01

In [212]:
df.to_csv(os.path.join(data_prep_path,'RIN_filtered_Combined.csv'))

In [27]:
from collections import Counter
Counter(df.DX)

Counter({nan: 7, 'CN': 118, 'Dementia': 56, 'MCI': 196})

In [28]:
df.columns

Index(['COLPROT', 'RIN', 'VISCODE', 'EXAMDATE', 'AGE', 'PTGENDER', 'PTEDUCAT',
       'PTETHCAT', 'PTRACCAT', 'APOE4',
       ...
       'AFFX-r2-TagH_at', 'AFFX-r2-TagIN-3_at', 'AFFX-r2-TagIN-5_at',
       'AFFX-r2-TagIN-M_at', 'AFFX-r2-TagJ-3_at', 'AFFX-r2-TagJ-5_at',
       'AFFX-r2-TagO-3_at', 'AFFX-r2-TagO-5_at', 'AFFX-r2-TagQ-3_at',
       'AFFX-r2-TagQ-5_at'],
      dtype='object', name='PTID', length=49418)

In [29]:
X = df.loc[:,'11715100_at':'AFFX-r2-TagQ-5_at']
X

PTID       11715100_at 11715101_s_at 11715102_x_at 11715103_x_at  \
Gene          HIST1H3G      HIST1H3G      HIST1H3G     TNFAIP8L1   
116_S_1249       2.237         2.624         1.873          2.92   
037_S_4410       2.294         2.416         1.884         2.668   
006_S_4153        2.14         2.322         1.999         3.634   
099_S_4205        2.04         2.395          2.08         3.278   
...                ...           ...           ...           ...   
052_S_0951       2.627         2.428         2.133         3.806   
014_S_4401       2.346         2.644         2.188         3.378   
036_S_4562       2.508         2.642         2.071         3.262   
018_S_4696       2.313         2.344         1.891           3.2   
128_S_2130       2.167         2.153         1.831          3.37   

PTID       11715104_s_at 11715105_at      11715106_x_at         11715107_s_at  \
Gene               OTOP2    C17ORF78  CTAGE6 || CTAGE15  F8A2 || F8A3 || F8A1   
116_S_1249         2.147       2.268              2.432                   4.6   
037_S_4410         2.156        2.13               2.27                 4.547   
006_S_4153         2.516       1.957              2.333                 4.442   
099_S_4205         2.251       2.154              2.334                 4.536   
...                  ...         ...                ...                   ...   
052_S_0951         2.423       2.073              2.345                 4.791   
014_S_4401         2.191       2.285              2.473                 4.442   
036_S_4562         1.893       2.283              2.325                 4.961   
018_S_4696         2.142       2.127              2.468                  4.33   
128_S_2130         2.246       2.042              2.539                 4.657   

PTID       11715108_x_at 11715109_at  ... AFFX-r2-TagH_at AFFX-r2-TagIN-3_at  \
Gene           LINC01098       SAMD7  ...             NaN                NaN   
116_S_1249         2.771       2.931  ...           2.355              2.624   
037_S_4410         2.363       2.548  ...             2.1               2.82   
006_S_4153         2.544       2.814  ...           2.165              2.455   
099_S_4205         2.546       2.864  ...           1.973              2.544   
...                  ...         ...  ...             ...                ...   
052_S_0951         2.383       2.722  ...           2.204              2.778   
014_S_4401         2.465       2.798  ...            2.21              2.291   
036_S_4562         2.645         2.5  ...           2.422              2.944   
018_S_4696         2.618       2.737  ...           2.157              2.622   
128_S_2130         2.359       2.793  ...           2.283              3.092   

PTID       AFFX-r2-TagIN-5_at AFFX-r2-TagIN-M_at AFFX-r2-TagJ-3_at  \
Gene                      NaN                NaN               NaN   
116_S_1249               2.01              2.906             2.463   
037_S_4410              1.726              2.465              2.26   
006_S_4153               1.84              2.681             2.251   
099_S_4205              1.909              2.548             2.266   
...                       ...                ...               ...   
052_S_0951              1.884              2.733             2.402   
014_S_4401              2.079              2.873             2.192   
036_S_4562              1.694              2.527             2.482   
018_S_4696              1.775              2.823             2.382   
128_S_2130              1.816              2.742             2.323   

PTID       AFFX-r2-TagJ-5_at AFFX-r2-TagO-3_at AFFX-r2-TagO-5_at  \
Gene                     NaN               NaN               NaN   
116_S_1249              2.05              2.06             1.858   
037_S_4410             1.933             1.717             2.208   
006_S_4153             1.985              1.77             2.184   
099_S_4205             2.077             1.838             2.085   


In [67]:
overall_median = X.iloc[1:].stack().median()
overall_median

3.898

In [75]:
filt_cols = []
probes = X.columns
for probe in probes:
    expr = X[probe][1:].astype(float)
    if sum(expr > overall_median) >= 100:
        filt_cols.append(probe)

In [76]:
len(filt_cols)

25845

In [77]:
X_filt = X.loc[:,filt_cols]
X_filt

PTID               11715107_s_at 11715110_at 11715113_x_at 11715114_x_at  \
Gene        F8A2 || F8A3 || F8A1      ARRDC5       FAM86C1       FAM86C1   
116_S_1249                   4.6        5.36         5.405         5.487   
037_S_4410                 4.547       5.031         4.703         4.746   
006_S_4153                 4.442       5.349         4.786         5.217   
099_S_4205                 4.536       5.111         4.578         4.562   
...                          ...         ...           ...           ...   
052_S_0951                 4.791       5.378         3.798         4.067   
014_S_4401                 4.442       4.529         4.213         4.152   
036_S_4562                 4.961       5.354         5.001         4.984   
018_S_4696                  4.33       4.577         5.064         5.245   
128_S_2130                 4.657       5.709         5.382         5.025   

PTID       11715122_at 11715125_at 11715129_s_at 11715132_x_at 11715135_s_at  \
Gene             RAB3D       MORC2        ZNF600     HIST1H2BH        SNAPC5   
116_S_1249       5.113       5.875         7.914          3.82         6.263   
037_S_4410       4.356       5.313         7.646         3.084         6.874   
006_S_4153       5.015       5.269         7.003         2.854         6.325   
099_S_4205           5       5.556         7.176         3.412         6.247   
...                ...         ...           ...           ...           ...   
052_S_0951       4.638       4.829         6.598         3.407         6.706   
014_S_4401       5.509       4.676         6.873         4.304         6.525   
036_S_4562       5.472       5.236         6.374         3.389         6.477   
018_S_4696       4.347       5.683         7.223         3.526         6.356   
128_S_2130       4.839       5.918         7.075         2.887         6.484   

PTID       11715144_s_at  ... AFFX-r2-Bs-thr-M_s_at AFFX-r2-Ec-bioB-3_at  \
Gene           KRTAP19-7  ...                   NaN                  NaN   
116_S_1249         3.256  ...                  8.21                7.892   
037_S_4410         3.404  ...                 7.494                7.647   
006_S_4153         4.112  ...                 8.061                8.455   
099_S_4205          3.67  ...                 7.508                8.024   
...                  ...  ...                   ...                  ...   
052_S_0951         3.867  ...                 8.828                8.491   
014_S_4401         3.588  ...                 8.292                7.974   
036_S_4562         3.193  ...                  8.66                8.326   
018_S_4696         3.712  ...                 7.842                8.357   
128_S_2130         3.938  ...                 8.703                8.726   

PTID       AFFX-r2-Ec-bioB-5_at AFFX-r2-Ec-bioB-M_at AFFX-r2-Ec-bioC-3_at  \
Gene                        NaN                  NaN                  NaN   
116_S_1249                8.023                8.196                9.238   
037_S_4410                7.724                7.962                8.521   
006_S_4153                8.519                 8.61                9.397   
099_S_4205                8.223                8.432                 9.12   
...                         ...                  ...                  ...   
052_S_0951                8.764                8.906                9.555   
014_S_4401                8.066                8.182                8.813   
036_S_4562                8.721                8.788                9.294   
018_S_4696                8.521                8.641                9.366   
128_S_2130                 8.98                9.207                9.784   

PTID       AFFX-r2-Ec-bioC-5_at AFFX-r2-Ec-bioD-3_at AFFX-r2-Ec-bioD-5_at  \
Gene                        NaN                  NaN                  NaN   
116_S_1249                9.396               11.419               11.514   
037_S_4410                9.204               11.627     

In [83]:
len(set(X_filt.columns))

25845

In [91]:
X_filt2 = X_filt.T

In [100]:
gene_expr = list(X_filt2[X_filt2['Gene'] == 'FAM86C1'].median())
len(gene_expr) #these many subjects

376

In [135]:
keys = set(X_filt2.Gene)

In [136]:
keys = [list(keys)[0]]+list(keys)[2:] # because keys[1] == nan

In [137]:
len(keys) #these many genes

11860

In [138]:
gene_dict = {key: None for key in keys}

In [140]:
for gene in keys:
    gene_dict[gene] = list(X_filt2[X_filt2['Gene'] == gene].median())

In [169]:
gene_df = pd.DataFrame(gene_dict)

In [170]:
gene_df['Subjects'] = list(X_filt.index[1:])

In [174]:
gene_df['DX'] = list(df['DX'][1:])

In [233]:
gene_df['AGE'] = list(df['AGE'][1:])
gene_df['PTGENDER'] = list(df['PTGENDER'][1:])
gene_df['APOE4'] = list(df['APOE4'][1:])
gene_df['PTEDUCAT'] = list(df['PTEDUCAT'][1:])

In [234]:
gene_df

RBMX2   COX6C  ENTPD5    DAK  ENSG00000184319  EEF1E1    IRF3  CD300E  \
0    7.4900  8.3010   6.109  4.705            3.753   4.931  7.3235   8.966   
1    7.7735  8.6605   5.453  4.292            3.536   6.025  7.0640   9.758   
2    7.8985  8.1230   6.055  4.672            3.623   4.061  7.4295   8.628   
3    7.6245  7.8755   5.879  4.554            4.739   4.639  7.2175   9.573   
4    7.8380  8.5340   6.368  4.797            4.432   4.421  7.3345   9.368   
..      ...     ...     ...    ...              ...     ...     ...     ...   
371  7.6165  7.8595   6.473  4.928            3.731   3.655  7.3250   9.873   
372  7.3045  7.2460   5.780  4.214            3.958   4.194  6.8910   9.480   
373  7.7035  8.0240   6.217  4.692            3.798   4.538  7.2105   9.649   
374  7.4770  8.1810   6.461  4.435            3.700   5.173  6.9080   9.714   
375  7.7640  7.9740   5.835  4.434            3.557   3.867  7.1810   9.731   

      EIF3H  FAM89B  ...  CNOT10     MR1  RPL31   DSN1    Subjects        DX  \
0    8.8255  8.3370  ...  7.1165  6.2515  7.619  5.949  116_S_1249        CN   
1    9.1160  8.0640  ...  7.1875  6.2065  7.043  6.043  037_S_4410        CN   
2    8.9465  8.0580  ...  7.1885  6.0585  6.747  5.126  006_S_4153  Dementia   
3    9.1370  8.2305  ...  7.2215  6.3030  6.671  5.777  099_S_4205       MCI   
4    9.0605  8.1515  ...  7.1900  6.1875  6.930  4.952  007_S_4467       MCI   
..      ...     ...  ...     ...     ...    ...    ...         ...       ...   
371  8.9925  8.2405  ...  7.2215  6.0680  6.007  4.991  052_S_0951        CN   
372  8.5600  8.2910  ...  7.1630  6.5115  6.353  4.960  014_S_4401        CN   
373  9.0650  8.2685  ...  7.4150  6.0485  6.685  4.673  036_S_4562       MCI   
374  8.7735  7.8810  ...  6.9165  6.2380  7.060  5.856  018_S_4696  Dementia   
375  9.2865  7.9810  ...  7.2805  6.1740  6.655  4.566  128_S_2130       MCI   

      AGE  PTGENDER  APOE4  PTEDUCAT  
0    70.8    Female      1        15  
1    69.1    Female      0        14  
2    79.3      Male      1        20  
3    81.4    Female      0        18  
4    61.9    Female      2        12  
..    ...       ...    ...       ...  
371  84.8      Male      0        18  
372  67.5    Female      1        20  
373  63.6      Male      2        18  
374    73    Female      1        16  
375  76.4      Male      0        16  

[376 rows x 11866 columns]

In [235]:
gene_df.to_csv(os.path.join(data_prep_path,'Preprocessed_gene_expr_dx.csv'))

In [201]:
X_filt2.to_csv(os.path.join(data_prep_path,'filtered_probes.csv'))

In [204]:
X_filt2[X_filt2['Gene']=='IRF3']

Gene 116_S_1249 037_S_4410 006_S_4153 099_S_4205 007_S_4467  \
PTID                                                                         
11718672_x_at  IRF3      7.355      7.252      7.401      7.236      7.528   
11745498_a_at  IRF3      8.664      8.403      8.728      8.499      8.737   
11745499_x_at  IRF3      7.292      7.105      7.458      7.231      7.296   
11753805_a_at  IRF3      4.582      4.426      4.748       4.57      4.907   
11753806_x_at  IRF3      3.744      3.594      4.211      3.962      3.987   
11756572_x_at  IRF3      7.419      7.023      7.523      7.204      7.373   

              128_S_0205 003_S_2374 031_S_2018 002_S_4654  ... 131_S_0384  \
PTID                                                       ...              
11718672_x_at      7.322      6.999      7.822      7.134  ...      7.962   
11745498_a_at      8.598      8.184      8.798      8.551  ...      9.069   
11745499_x_at      7.046      6.991      7.586      7.169  ...       7.68   
11753805_a_at      4.724      4.691      5.335      4.835  ...      4.991   
11753806_x_at      4.368      4.394      4.237      4.033  ...      4.396   
11756572_x_at      7.314      6.955       7.89      7.018  ...      7.757   

              135_S_4309 128_S_4571 099_S_2063 053_S_2357 052_S_0951  \
PTID                                                                   
11718672_x_at      7.228      7.052      7.333      6.287      7.497   
11745498_a_at      8.569      8.219       8.62      7.677      8.573   
11745499_x_at      7.317      7.147      7.349      6.498      7.308   
11753805_a_at       4.62      4.553       4.63      4.209      4.908   
11753806_x_at      4.128      4.201      3.662      3.517      4.177   
11756572_x_at      7.331      7.082      7.338      6.616      7.342   

              014_S_4401 036_S_4562 018_S_4696 128_S_2130  
PTID                                                       
11718672_x_at      6.965      7.329      6.987      7.307  
11745498_a_at      8.247      8.672      8.211      8.319  
11745499_x_at      6.817      7.092      6.872      7.077  
11753805_a_at       4.58      4.724      4.506      4.861  
11753806_x_at      3.776      4.021      3.618      4.268  
11756572_x_at      6.965       7.34      6.944      7.285  

[6 rows x 377 columns]

In [211]:
X_filt2[X_filt2['Gene']=='IRF3'].median()

116_S_1249    7.3235
037_S_4410    7.0640
006_S_4153    7.4295
099_S_4205    7.2175
007_S_4467    7.3345
               ...  
052_S_0951    7.3250
014_S_4401    6.8910
036_S_4562    7.2105
018_S_4696    6.9080
128_S_2130    7.1810
Length: 376, dtype: float64

In [216]:
list(df['DX'])[1:] == list(gene_df['DX'])

True

In [217]:
Counter(gene_df['DX'])

Counter({'CN': 118, 'Dementia': 56, 'MCI': 196, nan: 6})

# T-test on the preprocessed gene expression data (11860 genes)

In [218]:
df_CN = gene_df[gene_df['DX']=='CN']
df_AD = gene_df[gene_df['DX']=='Dementia']
df_MCI = gene_df[gene_df['DX']=='MCI']

In [220]:
df_CN_exp = df_CN.loc[:,'RBMX2': 'DSN1'].astype('float')
df_AD_exp = df_AD.loc[:,'RBMX2': 'DSN1'].astype('float')
df_MCI_exp = df_MCI.loc[:,'RBMX2': 'DSN1'].astype('float')

In [222]:
data = df_AD_exp['RBMX2'] #Normality test for just one example gene to be able to use student t test
from scipy import stats
stats.normaltest(data)
# null hypothesis: x comes from a normal distribution

NormaltestResult(statistic=1.0361857760527227, pvalue=0.5956554470716591)

In [223]:
from scipy.stats import ttest_ind
genes = df_AD_exp.columns
groups = ['AD_CN', 'AD_MCI', 'CN_MCI']
df_p = pd.DataFrame(columns=groups, index = genes) #datafrane for collecting p values for test
for gene in genes:
    data1 = df_AD_exp[gene]
    data2 = df_CN_exp[gene]
    stat, p = ttest_ind(data1, data2)
    df_p.loc[gene,'AD_CN'] = stat,p

for gene in genes:
    data1 = df_AD_exp[gene]
    data2 = df_MCI_exp[gene]
    stat, p = ttest_ind(data1, data2)
    df_p.loc[gene,'AD_MCI'] = stat, p
    
for gene in genes:
    data1 = df_CN_exp[gene]
    data2 = df_MCI_exp[gene]
    stat, p = ttest_ind(data1, data2)
    df_p.loc[gene,'CN_MCI'] = stat, p

    

In [225]:
df_fullstat = df_p 
df_fullstat

AD_CN  \
RBMX2             (-1.8253088263121886, 0.06968899075792705)   
COX6C             (-1.4526377359008553, 0.14814622301992758)   
ENTPD5            (-1.8105817077284156, 0.07195079505399883)   
DAK              (-1.9425407716326046, 0.053704760157490074)   
ENSG00000184319     (0.4099903338052268, 0.6823232342756071)   
...                                                      ...   
HMCES            (-2.5696310588622713, 0.011028837570413346)   
CNOT10            (-1.8138415347821064, 0.07144498916969112)   
MR1                 (1.1670547724982894, 0.2448036841861785)   
RPL31              (-0.03803751015009796, 0.969701887120763)   
DSN1              (-0.30586698247027616, 0.7600758121962519)   

                                                      AD_MCI  \
RBMX2               (0.0801913654846078, 0.9361492131880067)   
COX6C             (-0.20806158733592087, 0.8353502788990455)   
ENTPD5             (-1.5207750603530927, 0.1295800383247425)   
DAK                (-1.8291462848974513, 0.0685688316643392)   
ENSG00000184319   (-0.47422902374306836, 0.6357505294773624)   
...                                                      ...   
HMCES              (-1.5255397388849468, 0.1283885937648906)   
CNOT10             (-1.734146107550217, 0.08412485521446547)   
MR1              (-0.013498513392393535, 0.9892408374592925)   
RPL31               (0.4846654105685672, 0.6283380718237829)   
DSN1                (-0.670392127531587, 0.5032266563588108)   

                                                     CN_MCI  
RBMX2              (2.67388610275972, 0.007892615310077468)  
COX6C             (1.7720148240328097, 0.07736801991011391)  
ENTPD5             (0.5522604865410013, 0.5811652852205225)  
DAK                (0.4403746959814281, 0.6599706857163992)  
ENSG00000184319  (-1.2242187029325093, 0.22179333812547833)  
...                                                     ...  
HMCES              (1.1394249935069751, 0.2553999617937412)  
CNOT10             (0.2309273244488049, 0.8175226195924286)  
MR1               (-1.598688389065406, 0.11090202897822223)  
RPL31              (0.6423001637981411, 0.5211499738766587)  
DSN1             (-0.44705064140565176, 0.6551486708108709)  

[11860 rows x 3 columns]

In [226]:
from scipy.stats import ttest_ind
genes = df_AD_exp.columns
groups = ['AD_CN', 'AD_MCI', 'CN_MCI']
df_p = pd.DataFrame(columns=groups, index = genes) #datafrane for collecting p values for test
for gene in genes:
    data1 = df_AD_exp[gene]
    data2 = df_CN_exp[gene]
    stat, p = ttest_ind(data1, data2)
    df_p.loc[gene,'AD_CN'] = p

for gene in genes:
    data1 = df_AD_exp[gene]
    data2 = df_MCI_exp[gene]
    stat, p = ttest_ind(data1, data2)
    df_p.loc[gene,'AD_MCI'] = p
    
for gene in genes:
    data1 = df_CN_exp[gene]
    data2 = df_MCI_exp[gene]
    stat, p = ttest_ind(data1, data2)
    df_p.loc[gene,'CN_MCI'] = p



In [227]:
df_fullstat.to_csv(os.path.join(stats_path,'ttest_fullstat_filtered_genes.csv'))
df_p.to_csv(os.path.join(stats_path,'ttest_p_filtered_genes.csv'))